cloud/services/networkinterfaces/networkinterfaces.go

```diff
@@ -143,14 +163,74 @@ func (s *Service) Delete(ctx context.Context, spec interface{}) error {
 	}
 	klog.V(2).Infof("deleting nic %s", nicSpec.Name)
 	err := s.Client.Delete(ctx, s.Scope.ResourceGroup(), nicSpec.Name)
-	if err != nil && azure.ResourceNotFound(err) {
-		// already deleted
-		return nil
-	}
-	if err != nil {
+	if err != nil && !azure.ResourceNotFound(err) {
 		return errors.Wrapf(err, "failed to delete network interface %s in resource group %s", nicSpec.Name, s.Scope.ResourceGroup())
 	}
-
+	s.deleteInboundNatRule(ctx, nicSpec.PublicLoadBalancerName, s.MachineScope.Name())
 	klog.V(2).Infof("successfully deleted nic %s", nicSpec.Name)
 	return nil
 }
+
+func (s *Service) getInboundNatRule(ctx context.Context, lbName string, ruleName string) (network.InboundNatRule, error) {
+	lb, lberr := s.LoadBalancersClient.Get(ctx, s.Scope.ResourceGroup(), lbName)
+	if lberr != nil {
+		return network.InboundNatRule{}, lberr
+	}
+	for _, v := range *lb.InboundNatRules {
+		if to.String(v.Name) == ruleName {
+			return v, nil
+		}
+	}
+	return network.InboundNatRule{}, errors.Errorf("Inbound nat rule %s not found in load balancer %s", ruleName, lbName)
+}
+
+func (s *Service) createInboundNatRule(ctx context.Context, lbName string, rule network.InboundNatRule) error {
+	lb, lberr := s.LoadBalancersClient.Get(ctx, s.Scope.ResourceGroup(), lbName)
+	if lberr != nil {
+		return lberr
+	}
+	var sshFrontendPort int32 = 22
+	if len(*lb.InboundNatRules) != 0 {
+		ports := make(map[int32]struct{})
+		for _, v := range *lb.InboundNatRules {
+			if v.Name == rule.Name {
+				// Inbound Nat Rule already exists, nothing to do here.
+				return nil
+			}
+			ports[*v.InboundNatRulePropertiesFormat.FrontendPort] = struct{}{}
+		}
+		var i int32
+		found := false
+		for i = 2201; i < 2220; i++ {
```

Comment:
this means you could technically have 19 control planes... wasn't sure what to make this number. We can always adjust it later if there is a need for more control planes